flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/functions/FileCopyMapFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.api.functions;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.util.Preconditions;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * This mapper copies files from an existing savepoint into a new directory.
+ */
+public final class FileCopyMapFunction implements MapFunction<String, String> {
+
+	private static final long serialVersionUID = 1L;
+
+	// the destination path to copy file
+	private final String path;
+
+	public FileCopyMapFunction(String path) {
+		this.path = Preconditions.checkNotNull(path, "The destination path cannot be null");
+	}
+
+	@Override
+	public String map(String sourceFile) throws Exception {
+		// Create the destination directory before copying. It is not a problem if it exists already.
+		Path destPath = new Path(path);
+		destPath.getFileSystem().mkdirs(destPath);
+
+		Files.copy(
+			Paths.get(sourceFile), // source file
+			Paths.get(path, Paths.get(sourceFile).getFileName().toString()) // destination file
+		);
+		return sourceFile;
```

Comment:
This doesn't work, you're using java filesystem when you need to be using Flink's filesystem.

Reply:
Changed accordingly. Please have a look.